content/doc/book/using/remote-api.adoc

```diff
@@ -1,37 +1,280 @@
+filter.
+* Adding/removing downstream projects.
+* Chaining jobs i.e given a list of projects each project is added as a
+downstream project to the previous one.
+* Obtaining progressive console output.
+* Username/password based authentication.
+* Command Line Interface with a lot of options provided in the
+libraries.
+* Creating, listing views.
+* Adding jobs to views and removing jobs from views.
+* Adding/removing jenkins slaves, querying details of slaves.
+* Obtaining the tasks in build queue, and their age, cause, reason, ETA,
+ID, params and much more.
+* Quiet down, cancel quiet down, safe restart, force restart, and wait
+till Jenkins becomes available after a restart.
+* Ability to list installed/available plugins, obtain information about
+plugins, install/uninstall plugins and much more with plugins.
+
+This project is in rapid development and new features are getting added
+every day. Watch the progress
+https://github.com/arangamani/jenkins_api_client[here].
+
+[[RemoteaccessAPI-JavaAPIwrappers]]
+== Java API wrappers
+
+The https://github.com/cdancy/jenkins-rest[jenkins-rest] library is an
+object oriented java project that provides access to the Jenkins REST
+API programmatically to some remote API Jenkins provides. It is build
+using the amazing https://jclouds.apache.org/[jclouds toolkit] and can
+easily be extended to support more REST endpoints. Its feature set
+evolves and users are invited to contribute new endpoints via
+pull-requests. In its current state it is possible with this library to
+submit a job, track its progress through the queue, and during its
+execution until its completion, and obtain the build status. Services
+currently offered include:
 
-= Remote API
+* Endpoint definition (property or environment variable)
+* Authentication (basic and API token via property or environment
+variable)
+* Crumbs Issuer support (auto-detect crumbs)
+* Folder support
+* Jobs API (build, buildInfo, buildWithParameters, config, create,
+delete, description, disable, enable, jobInfo, lastBuildNumber,
+lastBuidTimestamp and progressiveText)
+* Plugin manager API (installNecessaryPlugins, list current plugins)
+* Queue API (cancel, list queue items, query queue item)
+* Statistics API (overall load)
+* Systems API (systemInfo)
 
-////
-Pages to mark as deprecated by this document:
+The project can evolve rapidly, this list is accurate only as of the
+date of writing.
 
-https://wiki.jenkins.io/display/JENKINS/Remote+access+API
-////
+[[RemoteaccessAPI-DetectingJenkinsversion]]
+== Detecting Jenkins version
 
-[NOTE]
-====
-This is still very much a work in progress
-====
+To check the version of Jenkins, load the top page (or, as of 1.483, any
+`+.../api/*+` page too) and check for the `+X-Jenkins+` response header.
+This contains the version number of Jenkins, like "1.404" This is also a
+good way to check if an URL is a Jenkins URL.
 
+[[RemoteaccessAPI-DiscoveringJenkinsonthenetwork]]
+== Discovering Jenkins on the network
```

Comment:
This has been removed from jenkins, no need to include it:
https://wiki.jenkins.io/display/JENKINS/Auto-discovering+Jenkins+on+the+network

> The content on this page only applies to Jenkins 2.219 and earlier. These features were disabled by default in Jenkins 2.219 and LTS 2.204.2, and removed entirely from Jenkins 2.200.

Reply:
ok, thanks @timja